moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -598,6 +600,36 @@ class KinematicsBase
   std::map<int, double> redundant_joint_discretization_;
   std::vector<DiscretizationMethod> supported_methods_;
 
+  /**
+   * @brief Enables kinematics plugins access to parameters that are defined
+   * for the 'robot_description_kinematics' namespace.
+   */
+  template <typename T>
+  inline bool lookupParam(const std::string& param, T& val, const T& default_val) const
+  {
+    ros::NodeHandle pnh("~");
+    if (pnh.hasParam(param))
+    {
+      val = pnh.param(param, default_val);
+      return true;
+    }
+
+    ros::NodeHandle nh;
+    if (nh.hasParam("robot_description_kinematics/" + param))
+    {
+      val = nh.param("robot_description_kinematics/" + param, default_val);
+      return true;
+    }
+
+    if (nh.hasParam("robot_description_kinematics/" + group_name_ + "/" + param))
+    {
+      val = nh.param("robot_description_kinematics/" + group_name_ + "/" + param, default_val);
```

Comment:
are we sure we want the ordering of looking for the parameter like this, with group second?

Reply:
No I'm not sure. I'm actually surprised to see it this way around now that you mention it...
In my opinion more specific places should go first and the order should be
- `~`
- `robot_description_kinematics/group_name/`
- `robot_description_kinematics`

@henhenhen could you please update the request (or provide insights that justify the current order)?

Reply:
Actually the order should have been like @v4hn mentioned. 
Private namespace as the most specific and prior default should be queried first.
Also subgroup should be checked before the group to allow group-wise specifications.

I changed it with commit https://github.com/ros-planning/moveit/pull/701/commits/53cb47064a49c13db9a6ba1524a27f91f63fb01b